microcosm_elasticsearch/searching.py

```diff
@@ -23,18 +25,45 @@ class SearchIndex:
         same type.
 
     """
-    def __init__(self, graph, index, model_class, doc_type=None):
+    __mapping_type_name__ = "doc"
+
+    @property
+    def mapping_type_name(self):
+        return self.__class__.__mapping_type_name__
+
+    @property
+    def doc_type_field(self):
+        """
+        Defines document field used to determine the document's polymorphic type in a single-mapping-type index
+
+        """
+        return "doctype"
+
+    def __init__(self, graph, index, doc_type=None):
         """
         :param graph: the object graph
         :param index: the name of an index to use
-        :param model_class: the model to return in the results list
         :param doc_type: the doc type to search for; uses the index's doc types if omitted
 
         """
         self.elasticsearch_client = graph.elasticsearch_client
         self.index = index
-        self.model_class = model_class
-        self.doc_type = doc_type
+        # Mapping from ES custom type field to corresponding model class
+        self.doc_types = dict()
+        self.mapping = Mapping(self.mapping_type_name)
+        if doc_type is not None:
+            self.register_doc_type(doc_type)
+
+    def register_doc_type(self, model_class):
```

Comment:
A `SearchIndex` holds polymorphic documents; each `Store` registers their `Model` with the index.